integration/spark2/src/main/scala/org/apache/spark/sql/optimizer/CarbonFilters.scala

```diff
@@ -78,9 +78,17 @@ object CarbonFilters {
           Some(new LessThanEqualToExpression(getCarbonExpression(name),
             getCarbonLiteralExpression(name, value)))
         case sources.In(name, values) =>
-          Some(new InExpression(getCarbonExpression(name),
-            new ListExpression(
-              convertToJavaList(values.map(f => getCarbonLiteralExpression(name, f)).toList))))
+          if (values.length == 1 && values(0) == null) {
+            Some(new InExpression(getCarbonExpression(name),
+              new ListExpression(
```

Comment:
Please make the In and Not In filter as par of TransformExpression. For e.g. Please look into line no 321-323..when it was corrected there it got missed in createFilter.

Reply:
I have made the null check for IN expression in line no. 89

Reply:
1. why two conditions length=1 and others when logic is same. only else condition should be sufficient
2. Not In case when null comes, in method TransformExpression (as sounak pointed), the behaviour was similar to falseExpression. So we have to conform not IN null case with hive to decide the behaviour.
3. Please add test cases for In and Not in case with null expression and compare behaviour with hive.

Reply:
The 1 condition was added when we have only NULL in our IN expression for that case, the Left expression was becoming NULL. So to avoid that we added this case.